diff --git a/node_modules/@hydrofoil/labyrinth/lib/request.js b/node_modules/@hydrofoil/labyrinth/lib/request.js
index 33bce51..a2cbd2e 100644
--- a/node_modules/@hydrofoil/labyrinth/lib/request.js
+++ b/node_modules/@hydrofoil/labyrinth/lib/request.js
@@ -15,6 +15,7 @@ function getRepresentation(req) {
 exports.getRepresentation = getRepresentation;
 function prefersMinimal(req) {
     const prefer = (0, parse_prefer_header_1.default)(req.header('Prefer'));
+    req.knossos.log("Prefer header: %o", prefer)
     return prefer.return === 'minimal';
 }
 exports.prefersMinimal = prefersMinimal;
